src/Carousel.tsx

```diff
@@ -265,6 +267,12 @@ function CarouselFunc(uncontrolledProps: CarouselProps, ref) {
     setRenderedActiveIndex(activeIndex || 0);
   }
 
+  useEffect(() => {
+    if (shouldClearNextDirection) {
+      nextDirectionRef.current = null;
+    }
```

Comment:
Tested the branch and this statement isn't being triggered for me.  It appears that `shouldClearNextDirection` is getting reset to `false` prior to running `useEffect`

Reply:
it's closing over the value from the most recent "render" pass. the issue is that strictmode is happier about discarding render results... so this is the side effect we need to put into `useEffect`

you can also repro the bug just by wrapping our example in a `<React.StrictMode>`

Reply:
and at least on my side, this fixes things

Reply:
I ran the example without strictmode in my case, so since the directionref isn't being cleared, skipping slides using the indicators on the bottom uses the wrong direction animation.  

Sliding on edges using left/right is fixed tho

Reply:
I think this is fixed now! Could you take another look?

Reply:
Using carousel indicators still moves in wrong direction :(

Attaching gif:

![carousel](https://user-images.githubusercontent.com/6854874/89445118-94d5bd80-d707-11ea-9a0c-add4c2838aae.gif)

Reply:
It seems to work for me: https://gfycat.com/annualtautgreatargus

How did you get that repro?

Reply:
Huh weird.  I just ran the docs locally from this branch.  Ran the uncontrolled example without strictmode

Windows 10
Chrome Version 84.0.4147.105
Firefox 79.0 (64-bit)

Reply:
I can't reproduce it on either Chrome or Firefox. Do you see it on https://deploy-preview-5322--react-bootstrap.netlify.app/components/carousel/ ?

Reply:
doh ignore.  my branch wasn't updated..  sorry!

strictmode and non strictmode works